src/services/brandService.ts

```diff
@@ -69,49 +69,117 @@ export const createBrandService = async (name: string, file: Express.Multer.File
-const downloadImageFromFirebase = async (fileName: string): Promise<Buffer> => {
-    // Create a reference to the file in Firebase Storage using the provided file path
+const getBrandImageUrlFromFirebase = async (imageUrl: string): Promise<string> => {
+    const fileName = imageUrl.split(`${bucket.name}/`)[1];
     const file = bucket.file(fileName);
+
+    try {
+        const [fileExists] = await file.exists();
+        if (fileExists) {
+        const [url] = await file.getSignedUrl({
+            action: 'read', 
+            expires: '03-09-2491' 
+        });
+        return url; 
+        }
+    } catch (error) {
+        console.error("Error fetching product image:", error);
+        return imageUrl; 
+    }
+    return imageUrl;
+};
+
+export const fetchBrandByIdService = async (id: string) => {
+    try {
+    const brand = await getBrandById(id);
+    if (!brand) {
+        return null;
+    }
+
+    const logoUrl = brand.logo;
+    const imageUrl = await getBrandImageUrlFromFirebase(logoUrl); 
+
+    return {
+        id: brand.id,
+        name: brand.name,
+        logo: imageUrl,
+    };
+
+    } catch (error) {
+    console.error('Error in fetching brand by ID:', error);
+    throw error; 
+    }
+};
+
+export const getAllBrandsService = async () => {
     try {
-        // fetching the file from storage
-        const [fileContents] = await file.download();
-        return fileContents;
+        const brands = await getAllBrandsRepo();
+
+        // process each brand to include signed image URL
+        const brandsWithLogos = await Promise.all(
+            brands.map(async (brand) => {
+                try {
+                    // get the signed URL for the image
+                    const imageUrl = await getBrandImageUrlFromFirebase(brand.logo);
+
+                    return {
+                        id: brand.id,
+                        name: brand.name,
+                        logo: imageUrl, // using the signed URL for the logo
+                    };
+                } catch (error) {
+                    console.warn(`Failed to fetch image for brand ID ${brand.id}:`, error);
+                    return {
+                        id: brand.id,
+                        name: brand.name,
+                        logo: null,
+                    };
+                }
+            })
+        );
+
+        return {
+            count: brandsWithLogos.length, // include the total count
+            brands: brandsWithLogos,       // the processed brand details
+        };
 
     } catch (error) {
+        console.error('Error in fetching all brands:', error);
+        throw new Error('Service error');
+    }
+};
 
-        console.error('Error downloading file from Firebase:', error);
-        throw new Error('Error during file download');
-        
+
+//  to delete an image from Firebase Storage
+export const deleteImageFromFirebase = async (fileName: string): Promise<void> => {
```

Comment:
this function also is it good to duplicate it in other services? Or is there a way to re-use the code?
@batooldshilleh  what do you think?